src/Dhall/Binary.hs

```diff
@@ -154,6 +154,12 @@ encode_1_1 (Const Type) =
     TString "Type"
 encode_1_1 (Const Kind) =
     TString "Kind"
+encode_1_1 (App (Merge _T₀ t₀) u₀) =
+    TList [ TInt 6, t₁, u₁, _T₁ ]
```

Comment:
This avoids breaking the encoding (and decoding) of the previous fully-applied `merge`.